apps/ledger-live-mobile/src/screens/Swap/SubScreens/SelectCurrency.tsx

```diff
@@ -32,22 +43,35 @@ export function SelectCurrency({
       // @ts-expect-error navigation type is only partially declared
       navigation.navigate(ScreenName.SwapForm, { currency });
     },
-    [track, navigation],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
   );
+
+  const RenderItem = useCallback(({ item }: { item: CryptoCurrency | TokenCurrency }) => {
+    return <CurrencyRow currency={item} onPress={onSelect} />;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   const sortedCurrencies = useCurrenciesByMarketcap(currencies);
 
   const renderList = useCallback(
     items => (
       <FlatList
         contentContainerStyle={styles.list}
+        removeClippedSubviews={true}
         data={items}
-        renderItem={({ item }) => <CurrencyRow currency={item} onPress={onSelect} />}
-        keyExtractor={currency => currency.id}
+        renderItem={RenderItem}
+        keyExtractor={keyExtractor}
```

Comment:
is currency.id not unique in this list?

Reply:
Ah, this was for testing ( had 1000's of the same currency ). Can revert this.